### DISCUSSION TITLE
Any Way to set up the ExHooks Before Startup ?

### DISCUSSION THREAD

**Discussion by TxMat** [emqx/emqx#12419] [Q&A] [answered]
Hi,

I'm running a clusterized EMQX Broker on AWS with etcd and for budget reason we shutdown the cluster when not at work with cause upon restart the loss of the ExHook configuration. 

We are working with custom EMQX docker images but as it is clustered i don't know if i modifying cluster.hocon directly will cause issues.

Thanks !

**Comment by SergeTupchiy** [comment, marked as answer]
Hi @TxMat, 
If any configuration is lost after cluster restart, then `cluster.hocon` was apparently not persisted. 
Modifying `cluster.hocon` may cause issues only if it was done with some errors. For example, if a manual change violates EMQX config schema, then the node will report the error and fail to start. 

You can create a necessary configuration, e.g. ExHook using Dashboard UI and then extract cluster.hocon, persist it and re-use when cluster restarts.
